pkg/api/v1/pod/util.go

```diff
@@ -261,9 +261,18 @@ func GetPodCondition(status *v1.PodStatus, conditionType v1.PodConditionType) (i
 	if status == nil {
 		return -1, nil
 	}
-	for i := range status.Conditions {
-		if status.Conditions[i].Type == conditionType {
-			return i, &status.Conditions[i]
+	return GetPodConditionFromList(status.Conditions, conditionType)
+}
+
+// GetPodConditionFromList extracts the provided condition from the given list of condition and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
```

Comment:
Comment is wrong.  See https://github.com/kubernetes/kubernetes/pull/64646#discussion_r192818782
and https://github.com/kubernetes/kubernetes/pull/64646#discussion_r192819412